static/js/components/AutoComplete.js

```diff
@@ -407,84 +420,43 @@ class AutoComplete extends Component {
+    if (maxSearchResults > 0) {
+      requestsList = requestsList.slice(0, maxSearchResults);
     }
 
-    dataSource.every((item, index) => {
-      switch (typeof item) {
-        case 'string':
-          if (filter(searchText, item, item)) {
-            requestsList.push({
-              text: item,
-              value: (
-                <MenuItem
-                  innerDivStyle={styles.innerDiv}
-                  value={item}
-                  primaryText={item}
-                  disableFocusRipple={disableFocusRipple}
-                  key={index}
-                />),
-            });
-          }
-          break;
-
-        case 'object':
-          if (item && typeof item.text === 'string') {
-            if (filter(searchText, item.text, item)) {
-              if (item.value.type && (item.value.type.muiName === MenuItem.muiName ||
-                 item.value.type.muiName === Divider.muiName)) {
-                requestsList.push({
-                  text: item.text,
-                  value: React.cloneElement(item.value, {
-                    key: index,
-                    disableFocusRipple: this.props.disableFocusRipple,
-                  }),
-                });
-              } else {
-                requestsList.push({
-                  text: item.text,
-                  value: (
-                    <MenuItem
-                      innerDivStyle={styles.innerDiv}
-                      primaryText={item.value}
-                      disableFocusRipple={disableFocusRipple}
-                      key={index}
-                    />),
-                });
-              }
-            }
-          }
-          break;
-
-        default:
-          // Do nothing
-      }
-
-      return !(maxSearchResults && maxSearchResults > 0 && requestsList.length === maxSearchResults);
-    });
+    let renderItem = option => (
+      <MenuItem
+        key={option.value}
+        primaryText={option.label}
+        value={option.value}
+        disableFocusRipple={disableFocusRipple}
+        innerDivStyle={styles.innerDiv}
+      />
+    );
 
     this.requestsList = requestsList;
+    let menuStyleWithHeight = Object.assign({}, menuStyle, {
+      maxHeight: menuHeight
+    });
 
     const menu = open && requestsList.length > 0 && (
       <Menu
         {...menuProps}
         ref="menu"
-        autoWidth={false}
+        autoWidth={true}
         disableAutoFocus={focusTextField}
         onEscKeyDown={this.handleEscKeyDown}
         initiallyKeyboardFocused={true}
         onItemTouchTap={this.handleItemTouchTap}
         onMouseDown={this.handleMouseDown}
-        style={Object.assign(styles.menu, menuStyle)}
+        style={Object.assign(styles.menu, menuStyleWithHeight)}
+        menuHeight={menuHeight}
         listStyle={Object.assign(styles.list, listStyle)}
-      >
-        {requestsList.map((i) => i.value)}
-      </Menu>
+        requestsList={requestsList}
```

Comment:
If we wanted to merge in future changes from material-ui it would be helpful to keep the existing variable names the same. I agree that it can be a tricky balance